docs/pallas/quickstart.md

```diff
@@ -0,0 +1,230 @@
+
++++
+
+So we've written what we call a "kernel". It's a program that will be executed on an accelerator. How do we invoke it from a JAX computation? We use the `pallas_call` higher-order function.
+
+```{code-cell} ipython3
+@jax.jit
+def add_vectors(x: jax.Array, y: jax.Array) -> jax.Array:
+  return pl.pallas_call(add_vectors_kernel,
+                        out_shape=jax.ShapeDtypeStruct(x.shape, x.dtype)
+                        )(x, y)
+add_vectors(jnp.arange(8), jnp.arange(8))
+```
+
+`pallas_call` takes in a kernel function as input and some other metadata about it. Here we specify `out_shape`, an object that has a `.shape` and `.dtype` (or a list thereof).
+`out_shape` determines the shape/dtype of `o_ref` in our `add_vector_kernel`.
+
+`pallas_call` returns a function that takes in and returns `jax.Array`s.
+
++++
+
+**What's actually happening here?**
+
+Thus far we've described how to think about Pallas kernels but what we've actually accomplished is we're writing a function that's executed very close to the compute units.
+
+On GPU, `x_ref` corresponds to a value in HBM and when we do `x_ref[...]` we are copying the value from HBM into SRAM (this is a costly operation generally speaking!). We then use GPU vector compute to execute the addition, then copy the resulting value in SRAM back to HBM.
+
+On TPU, we do something slightly different. Before the kernel is ever executed, we fetch the value from HBM into SRAM. `x_ref` therefore corresponds to a value in SRAM and when we do `x_ref[...]` we are copying the value from SRAM into a register. We then use TPU vector compute to execute the addition, then copy the resulting value back into SRAM. After the kernel is executed, the SRAM value is copied back into HBM.
+
++++
+
+## Pallas programming model
+
++++
+
+In our "hello world" example, we wrote a very simple kernel. It takes advantage of the fact that our 8-sized arrays can comfortably fit inside the SRAM of hardware accelerators. In most real-world applications, this will not be the case!
+
++++
+
+Part of writing Pallas kernels is thinking about how to take big arrays that live in high-bandwidth memory (HBM, also known as DRAM) and expressing computations that operate on "blocks" of those arrays that can fit in SRAM.
+
+### Grids
+
+To automatically "carve" up the inputs and outputs, you provide a `grid` and `BlockSpec`s to `pallas_call`.
+
+A `grid` is a tuple of integers (e.g. `()`, `(2, 3, 4)`, or `(8,)`) that specifies an iteration space.
+For example, a grid `(4, 5)` would have 20 elements: `(0, 0), (0, 1), ..., (0, 4), (1, 0), ..., (3, 4)`.
+We run the kernel function once for each element, a style of single-program multiple-data (SPMD) programming.
+
+<center>
+
+![A visualization of a 2D grid](grid.png)
```

Comment:
Once this is on the jax readthedocs, we might consider replacing the local url with a full http URL so the notebook works even when it's viewed outside JAX's directory structure.